projects/smoldot-browser-demo/src/index.ts

```diff
@@ -0,0 +1,61 @@
+// hack to make poladot-js work without bringing in webpack and babel
+import "regenerator-runtime/runtime"
+
+import { ApiPromise } from '@polkadot/api';
+import { SmoldotProvider }  from '@substrate/smoldot-provider';
+
+window.onload = () => {
+  const messages = document.getElementById('messages');
+  const messageHtml = (message: string) => {
+    const messageDiv = document.createElement('div');
+    messageDiv.classList.add('message');
+    messageDiv.appendChild(document.createTextNode(message));
+    return messageDiv;
+  };
+
+  const errorHtml = (message: string) => {
+    const messageDiv = document.createElement('div');
+    messageDiv.classList.add('message');
+    messageDiv.classList.add('error');
+    messageDiv.appendChild(document.createTextNode(message));
+    return messageDiv;
+  };
+
+  const displayMessage = (message: any) => {
+    if (messages === null) {
+      throw Error('No messages container.  Did you change the Html?');
+    }
+
+    messages.appendChild(message);
+  }
+
+  const error = (error: Error) => {
+    displayMessage(errorHtml(error.message));
+    throw error;
+  };
+
+  const log = (message: string) => {
+    displayMessage(messageHtml(message));
+  }
+
+  (async () => {
+    const response =  await fetch('./assets/westend.json')
+    if (!response.ok) {
+      error(new Error('Error downloading chain spec'));
+    }
+
+    const chainSpec =  await response.text();
```

Comment:
Minor nitpick: I don't think this needs an await, because `response` is already loaded async

Reply:
Nope, I thought that too: https://developer.mozilla.org/en-US/docs/Web/API/Body/text